### PR TITLE
feat(panel): Add clickOutsideToClose and escapeToClose config options to the panel.

### PR DESCRIPTION
**Pull request by KarenParker**
@gmoothart @ErinCoughlan @jelbourn @DerekLouie

Please review.

Fixes issues #7195 and #7194.

### CODE REVIEW

File: src/components/panel/panel.js [hunk 1]
```diff
@@ -662,6 +667,7 @@ MdPanelRef.prototype.attach = function() {
   this._attachPromise = this._$q(function(resolve, reject) {
     self._createPanel().then(function() {
       self.isAttached = true;
+      self._activeListeners();
```
Comment (jelbourn) [inline src/components/panel/panel.js]: `_addEventListeners()` ?

File: src/components/panel/panel.js [hunk 2]
```diff
@@ -832,6 +838,82 @@ MdPanelRef.prototype._addStyles = function() {
...
+
+    // Queue remove listeners function
+    removeListeners.push(function() {
+      target.off('mousedown', mousedownHandler);
+      target.off('mouseup', mouseupHandler);
+    });
+  }
+
+  // Attach specific `remove` listener handler
+  this._deactivateListeners = function() {
```
Comment (jelbourn) [inline src/components/panel/panel.js]: This should just be a separate `prototype` method.

File: src/components/panel/panel.js [hunk 3]
```diff
@@ -832,6 +838,82 @@ MdPanelRef.prototype._addStyles = function() {
 };
 
 
+/**
+ * Listen for escape keys and outside clicks to auto close.
+ * @private
+ */
+MdPanelRef.prototype._activeListeners = function() {
```
Comment (jelbourn) [inline src/components/panel/panel.js]: How about breaking this into two methods like `configureEscapeToClose` and `configureClickOutsideToClose` ?

**Comment by KarenParker** [comment]
Addressed comments.

**Comment by DerekLouie** [comment]
LGTM

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**